adam/visualization/positioning.py

```diff
@@ -0,0 +1,371 @@
+from itertools import combinations
```

Comment:
Try to rearrange the things in this module in the order you'd like a reader to encounter them (as much as Python will allow)